### PR TITLE
[SPARK-26035][PYTHON] Break large streaming/tests.py files into smaller files

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR continues to break down a big large file into smaller files. See https://github.com/apache/spark/pull/23021. It targets to follow https://github.com/numpy/numpy/tree/master/numpy.

Basically this PR proposes to break down `pyspark/streaming/tests.py` into ...:

```
pyspark
├── __init__.py
...
├── streaming
│   ├── __init__.py
...
│   ├── tests
│   │   ├── __init__.py
│   │   ├── test_context.py
│   │   ├── test_dstream.py
│   │   ├── test_kinesis.py
│   │   └── test_listener.py
...
├── testing
...
│   ├── streamingutils.py
...
```

## How was this patch tested?

Existing tests should cover.

`cd python` and .`/run-tests-with-coverage`. Manually checked they are actually being ran.

Each test (not officially) can be ran via:

```bash
SPARK_TESTING=1 ./bin/pyspark pyspark.tests.test_context
```

Note that if you're using Mac and Python 3, you might have to `OBJC_DISABLE_INITIALIZE_FORK_SAFETY=YES`.

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
I haven't tested the kinesis logic yet. I will check it via Jenkins.

Line counts:

```
     751 ./test_dstream.py
      89 ./test_kinesis.py
     158 ./test_listener.py
     184 ./test_context.py
```

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by HyukjinKwon** [comment]
Will go and merge this tomorrow if there's no outstanding issues.

cc @zsxwing and @tdas.

File: python/pyspark/testing/streamingutils.py [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+    else:
+        return jars[0]
+
+
+# Must be same as the variable and condition defined in KinesisTestUtils.scala and modules.py
+kinesis_test_environ_var = "ENABLE_KINESIS_TESTS"
+should_skip_kinesis_tests = not os.environ.get(kinesis_test_environ_var) == '1'
+kinesis_asl_assembly_jar = search_kinesis_asl_assembly_jar()
+
+if should_skip_kinesis_tests:
```
Comment (HyukjinKwon) [inline python/pyspark/testing/streamingutils.py]: I simplified this logic and tested each if-else branch.

**Comment by HyukjinKwon** [comment]
Merged to master.

**Comment by HyukjinKwon** [comment]
@BryanCutler, should be ready to work on ML and MLlib as well.

**Comment by HyukjinKwon** [comment]
Also, @BryanCutler, I think we can talk about locations of `testing/...util.py` later when we finished to split the tests. Moving utils would probably cause less conflicts and should be good enough to separately discuss if that's a worry, and should be changed.

**Comment by BryanCutler** [comment]
> Also, @BryanCutler, I think we can talk about locations of testing/...util.py later when we finished to split the tests. Moving utils would probably cause less conflicts and should be good enough to separately discuss if that's a worry, and should be changed.

Sounds good, working on MLlib right now. Hopefully have a PR up soon

**Comment by HyukjinKwon** [comment]
Thank you @BryanCutler.

**Comment by zsxwing** [comment]
LGTM. By the way, @HyukjinKwon I totally understand that this PR needs to merge soon to avoid getting conflicts. But could you please at least get someone to review and sign off before merging next time? Thanks!

**Comment by HyukjinKwon** [comment]
@zsxwing sure. Sorry that I rushed. Will do next time.

**Pull request closed without merging**